### DISCUSSION TITLE
Creating an app shell and a default "No internet connection page" fallback

### DISCUSSION THREAD

**Discussion by ivandotv** [shadowwalker/next-pwa#163] [Q&A] [answered]
This package does a great job of caching the routes, however, in a true PWA we also need an **offline** mode. So when the user is offline and it navigates to a page that is *not cached* it should display something. 
There is a great article about it on [web fundamentals site](https://developers.google.com/web/fundamentals/architecture/app-shell).
So basically we need a **default fallback** for every route that is not cached.
A great example of this functionality is https://dev.to/ if you install this site as PWA , switch your phone to **airplane** mode,  and try to refresh the app or navigate somewhere, it will show the default fallback page instead of the default browser "no internet connection" page.

Any ideas?

**Comment by shadowwalker** [comment, marked as answer]
Sure, the feature is implemented.

To get started simply add a `/_offline` page such as `pages/_offline.js` or `pages/_offline.jsx` or `pages/_offline.ts` or `pages/_offline.tsx`. Then you are all set! When the user is offline, all pages which are not cached will fallback to '/_offline'.

**[Use this example to see it in action](https://github.com/shadowwalker/next-pwa/tree/master/examples/offline-fallback-v2)**s

**Reply by ivandotv** [reply]
This feature wasn't available when I asked the question :)
Great example, thank you for all your hard work.
